WooCommerce/Classes/ViewRelated/Orders/Order Creation/Synchronizer/ProductInputTransformer.swift

```diff
@@ -41,6 +41,58 @@ struct ProductInputTransformer {
         return order.copy(items: items)
     }
 
+    /// Adds, deletes, or updates order items based on the multiple given product inputs
+    ///
+    /// - Parameters:
+    ///   - input: Array of types of products the synchronizer supports
+    ///   - order: Represents an Order entity.
+    ///   - updateZeroQuantities: When true, items with `.zero` quantities will be updated instead of being deleted.
+    ///
+    /// - Returns: An Order entity.
+    static func updateMultiple(input: [OrderSyncProductInput], on order: Order, updateZeroQuantities: Bool) -> Order {
+        var items = order.items
+
+        for productInput in input {
+            items.append(contentsOf: updateOrderItems(
+                from: productInput,
+                order: order,
+                updateZeroQuantities: updateZeroQuantities)
+            )
+        }
+
+        return order.copy(items: items)
+    }
+
+    /// Updates the `OrderItems` array with `OrderSyncProductInput`.
+    /// Uses the same implementation as `update()` but returns an array of OrderItems instead,
+    /// rather than aggregating them into the Order
+    ///
+    /// - Parameters:
+    ///   - input: Types of products the synchronizer supports
+    ///   - order: Represents an Order entity.
+    ///   - updateZeroQuantities: When true, items with `.zero` quantities will be updated instead of being deleted.
+    /// - Returns: An array of Order Item entities
+    private static func updateOrderItems(from input: OrderSyncProductInput,
+                                         order: Order, updateZeroQuantities: Bool) -> [OrderItem] {
+        // If the input's quantity is 0 or less, delete the item if required.
+        guard input.quantity > 0 || updateZeroQuantities else {
```

Comment:
> This is a perfect example where we can create a different function just for that, or call

That makes sense, I've re-used the existing `remove()` method on https://github.com/woocommerce/woocommerce-ios/pull/9173/commits/7ca36761c6405905e07d0438188c367a9365ecce as I now realized I only needed to append it with `.items` to get the exact returning type we wanted.

Reply:
> Furthermore, `updateZeroQuantities` is not expressive enough, I couldn't expect the item is removed by setting that property to true

I agree with `updateZeroQuantities` being a bit hard to read, this was part of the existing logic for single-product update() method so I didn't want to modify it as part of this PR, I just re-used it. I'm considering using an `Enum` with `update` and `delete` cases, and change the method signatures to something like the following:

```
    /// Used to represent if a `.zero` quantity item should be updated or deleted
    ///
    enum ZeroQuantityItemAction {
        case update
        case delete
    }

    /// Adds, deletes, or updates order items based on the given product input.
    ///
    static func update(input: OrderSyncProductInput, on order: Order, shouldUpdateOrZeroQuantityItem shouldUpdateOrDelete: ZeroQuantityItemAction) -> Order {
        // If the input's quantity is 0 or less, delete the item if required.
        guard input.quantity > 0 || shouldUpdateOrDelete == .update else {
            return remove(input: input, from: order)
        }
```

What do you think? Despite being clearer, I still believe the naming is a bit hard to read 🤔

Reply:
Thanks for letting me know! As it was part of an existing function, it's ok to leave it for now. But it would be great to create an issue so we can address it in the future.

Reply:
Definitely, created here: https://github.com/woocommerce/woocommerce-ios/issues/9201 as part of this project.